pkg/apis/devconsole/v1alpha1/gitsourceanalysis_types.go

```diff
@@ -0,0 +1,84 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// GitSourceAnalysisSpec defines the desired state of GitSourceAnalysis
+type GitSourceAnalysisSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
+
+	// GitSourceRef refers to the GitSource to be analyzed
+	GitSourceRef GitSourceRef `json:"gitSourceRef"`
+}
+
+// GitSourceRef refers to the GitSource to be analyzed
+type GitSourceRef struct {
+	// Name is the name of the GitSource that contains all necessary information of the git repo
```

Comment:
Worth mentioning that it's the name of the GitSource **in the same namespace** unless you want to be able to refer to GitSources from another namespace. But in that case we would need to add `namespace` field too.

Reply:
good point. changed https://github.com/redhat-developer/devconsole-api/pull/8/files#diff-40860617e11838e34856b8f57fc5cda7R20
thanks